### ISSUE TITLE
Multiple items in the options menu shorten the text width

### ISSUE THREAD

**Issue by PattaFeuFeu** [opacapp/multiline-collapsingtoolbar#26]
Once there is a certain amount of items in the options menu, the toolbar text width decreases. On a 1080p xxhdpi phone, more than two items lead to a massive shortage, depending on the text of course.

Attached, you will find screenshots taken with the demo app running and the following changes introduced. 

To reproduce, paste the following into menu_demo.xml:

```xml
<?xml version="1.0" encoding="utf-8"?>
<menu xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto">
    <item
        android:id="@+id/action_settings0"
        android:icon="@android:drawable/btn_star"
        android:title="@string/settings"
        app:showAsAction="ifRoom" />
    <item
        android:id="@+id/action_settings1"
        android:icon="@android:drawable/btn_star"
        android:title="@string/settings"
        app:showAsAction="ifRoom" />
    <item
        android:id="@+id/action_settings2"
        android:icon="@android:drawable/btn_star"
        android:title="@string/settings"
        app:showAsAction="ifRoom" />

</menu>
```

### Zero items:

![zero_items](https://cloud.githubusercontent.com/assets/9012887/20100319/454e2812-a5bd-11e6-87f8-f85b5f8ed842.png)

### One item:

![one_item](https://cloud.githubusercontent.com/assets/9012887/20100323/454f449a-a5bd-11e6-830b-3c9ee150f48c.png)

### Two items:

![two_items](https://cloud.githubusercontent.com/assets/9012887/20100320/454e706a-a5bd-11e6-8374-17fcdcfaf650.png)

### Three items:

![three_items](https://cloud.githubusercontent.com/assets/9012887/20100321/454e9176-a5bd-11e6-9964-58bb384b05e4.png)

### Four items:

![four_items](https://cloud.githubusercontent.com/assets/9012887/20100322/454f09e4-a5bd-11e6-8703-2a4113fd7f22.png)

**Comment by johan12345** [comment]
Thanks for reporting the issue!

I think it is caused by a change introduced with the Support Library 24.1.1 to 24.2.1 update. I think before it would just move the additional options menu items to the overflow menu if the text was too long.

The problem is that if we want to fix this properly we need to make sure that the title has the correct width in the collapsed state (staying to the left of the icons) and a different width in the expanded state, allowing for additional words in the first line.

This should theoretically be doable, but I am currently quite busy and therefore probably not able look into it very soon. Pull requests are welcome :)

**Comment by TonyHaddad91** [comment]
@PattaFeuFeu have u fixed ur issue?

**Comment by micookie** [comment]
I have  the same problem.

I think expanded Title width  used collapsed title's width....  it should re-measure while expanded .

**Comment by boformer** [comment]
This is still an issue.

**Comment by SimonSchubert** [comment]
#46 I uploaded a pull requested with a fix. It works like a charm for me and hopefully for you guys too. 
![device-2017-10-23-153359](https://user-images.githubusercontent.com/1833337/31891847-d40b002c-b807-11e7-87d9-ece27bb769da.png)
![device-2017-10-23-153409](https://user-images.githubusercontent.com/1833337/31891857-da6f6cd2-b807-11e7-80b1-a59419dbb777.png)
![device-2017-10-23-153424](https://user-images.githubusercontent.com/1833337/31891866-e2df21f0-b807-11e7-9945-1db0edfe4843.png)

**Comment by johan12345** [comment]
This should be fixed through PR #46 in the next release.
@raphaelm, could you make one when you are back from your journey?

**Comment by johan12345** [comment]
Ah, and don't forget to take into account #52 when making the new release!

**Comment by johan12345** [comment]
The fix from #46 is included in the newest release `27.1.0`. Thanks @SimonSchubert !
